docs/data-sources/dms_kafka_flavors.md

```diff
@@ -0,0 +1,153 @@
+---
+subcategory: "Distributed Message Service (DMS)"
+---
+
+# huaweicloud_dms_kafka_flavors
+
+Use this data source to get the list of available flavor details within HuaweiCloud.
+
+## Example Usage
+
+### Query the list of kafka flavors that support the prepaid charging mode
+
+```hcl
+data "huaweicloud_dms_kafka_flavors" "test" {
+  charging_mode = "hourly"
+}
+```
+
+### Query the list of kafka flavors for cluster type
+
+```hcl
+data "huaweicloud_dms_kafka_flavors" "test" {
+  type = "cluster"
+}
+```
+
+### Query the kafka flavor details of the specified ID
+
+```hcl
+data "huaweicloud_dms_kafka_flavors" "test" {
+  flavor_id = "c6.2u4g.cluster"
+}
+```
+
+### Query list of kafka flavors that available in the availability zone list
+
+```hcl
+variable "az1" {}
+variable "az2" {}
+
+data "huaweicloud_dms_kafka_flavors" "test" {
+  availability_zones = [
+    var.az1,
+    var.az2,
+  ]
+}
+```
+
+## Argument Reference
+
+* `region` - (Optional, String) Specifies the region in which to obtain the dms kafka flavors.
+  If omitted, the provider-level region will be used.
+
+* `flavor_id` - (Optional, String) Specifies the DMS flvaor ID, e.g. **c6.2u4g.cluster**.
+
+* `storage_spec_code` - (Optional, String) Specifies the disk IO encoding.
+  + **dms.physical.storage.high.v2**: Type of the disk that uses high I/O.
+  + **dms.physical.storage.ultra.v2**: Type of the disk that uses ultra-high I/O.
+
+* `type` - (Optional, String) Specifies flavor type. The valid values are **single** and **cluster**.
+
+* `arch_type` - (Optional, String) Specifies the type of CPU architecture, e.g. **X86**.
+
+* `availability_zones` - (Optional, List) Specifies the list of availability zones with available resources.
+
+* `charging_mode` - (Optional, String) Specifies the flavor billing mode.
+  The valid valus are as follows:
+  + **monthly**: Pre-paid mode.
```

Comment:
please consider the value the this parameter, how about keep it same with other resources

Reply:
done.